### ISSUE TITLE
Export cephfs/directory's list() and names() functions

### ISSUE THREAD

**Issue by isi-lincoln** [ceph/go-ceph#327]
I've opened a PR which exports list() and names() from cephfs/directory. 

https://github.com/ceph/go-ceph/pull/326

I think that they functions would be of much more use to the community as exported functions.  They dont modify any private fields, and they arent used internally anywhere other than in the tests.

Thoughts/Comments?

Thank you.

**Comment by phlogistonjohn** [comment]
We'll certainly consider it. The conversation can take place in the PR.

(meta-comment) This project doesn't require you to have an issue for PRs so feel free in the future to just file the PR. There's nothing wrong with filing an issue first for a topic but if you end up filing both at (or around) the same time you should not feel obligated to do so.

**Comment by phlogistonjohn** [comment]
@isi-lincoln any update on this stuff? No rush, just curious. Thanks!

**Comment by isi-lincoln** [comment]
I have fallen off the cliff on this.  I apologize.  If someone else is able to take this over that would be greatly appreciated.

**Comment by phlogistonjohn** [comment]
No worries, and thank you for the update. I think we can leave the issue open but if you don't think you'll progress on the PR you may want to close it. There's nothing stopping us from reopening it in the future or you filing a new PR.

I just checked and it appears that the Go language proposal for a fs abstraction package has [been accepted](https://github.com/golang/go/issues/41190) and [is merged](https://github.com/golang/go/blob/master/src/io/fs/fs.go) in the Go source tree. I don't pay close attention to the Go development process but I suspect this means that it will be released in Go 1.16. In light of this, I suggest we wait on this issue until Go 1.16 is out and then we can update or extend our code once we can read the final documentation and write tests against the official interface types.

**Comment by phlogistonjohn** [comment]
Thanks, I think we can safely close this. If additional APIs are still desired the issue can be reopened or a new issue filed.

Closing.
